hangups/client.py

```diff
@@ -149,6 +149,14 @@ def disconnect(self):
         When disconnection is complete, :func:`connect` will return.
         """
         logger.info('Disconnecting gracefully...')
+
+        # ignore a previous Exception
+        # pylint:disable=protected-access
+        if self._listen_future._exception is not None:
+            logger.info('disconnect: discard %s',
+                        repr(self._listen_future._exception))
+            self._client._listen_future._exception = None
```

Comment:
A broad except is needed then. https://github.com/tdryer/hangups/pull/345/commits/dde895c882e9df3b8241a59c3052c33608c88a99 solves this without a raised Exception.